### PR TITLE
ROC-7993: Framework for rendering template

### PR DESCRIPTION
**Pull request by daveoh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7803

### Change description ###
framework for rendering template

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderRenderer.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    public DocAssemblyResponse renderLegalAdvisorOrder(CCDCase ccdCase, String authorisation) {
+        return renderOrder(ccdCase, authorisation, legalAdvisorTemplateId);
+    }
+
+    public DocAssemblyResponse renderJudgeOrder(CCDCase ccdCase, String authorisation) {
+        return renderOrder(ccdCase, authorisation, judgeTemplateId);
+    }
+
+    public DocAssemblyResponse renderOrder(CCDCase ccdCase, String authorisation) {
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderRenderer.java:45]: should it be bit upper?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java [hunk 2]
```diff
@@ -163,6 +168,31 @@ public DocAssemblyTemplateBody changeContactBody(CCDCase ccdCase) {
...
     }
 
+    public DocAssemblyTemplateBody paperDefenceForm(CCDCase ccdCase) {
+        CCDRespondent defendant = ccdCase.getRespondents().get(0).getValue();
+        String defendantName = defendant.getClaimantProvidedPartyName();
+        CCDAddress defendantAddress = getDefendantAddress(defendant);
+
+        CCDApplicant claimant = ccdCase.getApplicants().get(0).getValue();
+
+        LocalDate extendedResponseDeadline =
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java:178]: I think response deadline is reset here, not extended, so it be updated in ccdcase and picked from there

Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java:178]: Will double check the requirements but it is my understanding that we need both the original deadline and the extended. But you are correct in pointing out that this date may have already been extended.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderRenderer.java (hunk 1 above)
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderRenderer.java:45]: Do you mean that the order of the methods should be rearranged?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java (hunk 2 above)
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java:178]: Actually as the form will get generated when the claimant submits the claim then the defendant will not have had a chance to extend the deadline.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderRenderer.java (hunk 1 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderRenderer.java:45]: yes

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java (hunk 2 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java:178]: Form has option to extend deadline, so they can return part of the form with new deadline request

**Pull request merged**